cli/src/pcluster/models/imagebuilder.py

```diff
@@ -12,160 +12,112 @@
-        self.components = components
-
-    def _validate(self):
-        self._execute_validator(
-            InstanceTypeBaseAMICompatibleValidator,
-            instance_type=self.instance_type,
-            image=self.parent_image,
-        )
-
-
-# ---------------------- Dev Settings ---------------------- #
-
-
-class ImagebuilderDevSettings(BaseDevSettings):
-    """Represent the dev settings configuration for the ImageBuilder."""
-
-    def __init__(
-        self,
-        update_os_and_reboot: bool = None,
-        disable_pcluster_component: bool = None,
-        distribution_configuration: DistributionConfiguration = None,
-        terminate_instance_on_failure: bool = None,
-        **kwargs
-    ):
-        super().__init__(**kwargs)
-        self.update_os_and_reboot = Resource.init_param(update_os_and_reboot, default=False)
-        self.disable_pcluster_component = Resource.init_param(disable_pcluster_component, default=False)
-        self.distribution_configuration = distribution_configuration
-        self.terminate_instance_on_failure = Resource.init_param(terminate_instance_on_failure, default=True)
-
-
-# ---------------------- ImageBuilder ---------------------- #
-
-
-class ImageBuilder(Resource):
-    """Represent the configuration of an ImageBuilder."""
-
-    def __init__(
-        self,
-        image: Image,
-        build: Build,
-        dev_settings: ImagebuilderDevSettings = None,
-    ):
-        super().__init__()
-        self.image = image
-        self.build = build
-        self.dev_settings = dev_settings
-
-    def _validate(self):
-        self._execute_validator(
-            EbsVolumeTypeSizeValidator,
-            volume_type=ROOT_VOLUME_TYPE,
-            volume_size=self.image.root_volume.size,
-        )
-        self._execute_validator(
-            AMIVolumeSizeValidator,
-            volume_size=self.image.root_volume.size,
-            image=self.build.parent_image,
-        )
+import json
+import logging
+
+from common.aws.aws_api import AWSApi
+from common.aws.aws_resources import Stack
+from common.boto3.common import AWSClientError
+from common.imagebuilder_utils import ImageBuilderStatusMapping
+from pcluster.models.common import BaseTag
+from pcluster.schemas.imagebuilder_schema import ImageBuilderSchema
+from pcluster.templates.cdk_builder import CDKTemplateBuilder
+from pcluster.utils import get_installed_version
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ImageBuilderActionError(Exception):
+    """Represent an error during the execution of an action on the imagebuilder."""
+
+    def __init__(self, message: str, validation_failures: list = None):
+        super().__init__(message)
+        self.validation_failures = validation_failures or []
+
+
+class ImageBuilderStack(Stack):
+    """Class representing a running stack associated to a building image."""
+
+    def __init__(self, name: str):
+        """Init stack info."""
+        super().__init__(name)
+        try:
```

Comment:
Please take a look at https://github.com/aws/aws-parallelcluster/pull/2483

I changed this initialization, by moving the describe_stack in a property.
This helps when mocking the object and when istantiating it directly.